libsolidity/analysis/DocStringAnalyser.cpp

```diff
@@ -48,9 +48,37 @@ void copyMissingTags(StructurallyDocumentedAnnotation& _target, set<CallableDecl
 	for (auto const& iterator: _target.docTags)
 		existingTags.insert(iterator.first);
 
-	for (auto const& [tag, content]: sourceDoc.docTags)
-		if (tag != "inheritdoc" && !existingTags.count(tag))
-			_target.docTags.emplace(tag, content);
+	for (auto it = sourceDoc.docTags.begin(); it != sourceDoc.docTags.end();)
+	{
+		size_t n = 0;
+		for (auto next = sourceDoc.docTags.upper_bound(it->first); it != next; it++, n++)
+		{
+			if (it->first == "inheritdoc" || existingTags.count(it->first))
+				continue;
+
+			DocTag newContent = it->second;
+			newContent.inherited = true;
+
+			if (_callable && it->first == "return")
+			{
+				size_t docParaNameEndPos = it->second.content.find_first_of(" \t");
+				string const docParameterName = it->second.content.substr(0, docParaNameEndPos);
+
+				if (docParameterName != _callable->returnParameters().at(n)->name())
+				{
+					bool baseHasNoName = (*_baseFunctions.begin())->returnParameters().at(n)->name().empty();
+					newContent.content =
+						_callable->returnParameters().at(n)->name() + (
+							string::npos == docParaNameEndPos || baseHasNoName ?
+							" " + newContent.content :
+							newContent.content.substr(docParaNameEndPos)
+						);
+				}
```

Comment:
Maybe we can replace this by

```cpp
regex_replace(
     it->second.content,
     regex{"[^\\s\\t]"},
     _callable->returnParameters().at(n)->name(),
     regex_constants::format_first_only
)
```

Reply:
Hmm generally I try to avoid the complexity of regex when I don't really need it.